### DISCUSSION TITLE
Looking for a NA compatible 3-way switch

### DISCUSSION THREAD

**Discussion by SylvainGa** [arendst/Tasmota#20223] [Q&A]
Hi, after several years of faithful service, one of my Moe's SS01-1 switch flashed with Tasmota gave up. Having not looked for a Tasmota compatible switch in a few years, I'm not sure where to start looking. blakadder.com seems to be out of date. Anybody has any pointers for me or know a 3-way switch that can still be flashed with Tasmota available in Canada?

Thanks.

**Comment by SylvainGa** [comment]
Well, that's not reassuring about the future of Tasmota on cheap chinese switches.

**Reply by sfromis** [reply]
There are also some devices with factory flashed Tasmota, and recently devices have come out with ESP32 which is great for Tasmota (unless fused shut for flashing).

**Reply by barbudor** [reply]
Sonoff, Shelly are still actively using ESP chips and have manufacturing quality slighly higher that the cheapest chinese brands.
Athom has Tasmota pre-flashed devices but I believe they are only EU devices.
May be moving away from the lower-end devices is not a bad idea ?

**Reply by SylvainGa** [reply]
Hi, do you know of one that isn't "fused shot"?

**Reply by SylvainGa** [reply]
Sonoff don't make any that fits in a box with two other switches and Shelly are too big to be mixed in a box with other switches, especially if that module needs be used as a three way switch, which needs more than one 'button'. Cheap Chinese one at least did the trick.

**Reply by sfromis** [reply]
I'd certainly not expect devices on https://templates.blakadder.com/esp32.html to have an eFuse setup preventing flashing, but my note was not to suggest that such is common, more to mention that having ESP32 does not automatically mean open to flashing, unlike ESP8266.

**Reply by sfromis** [reply]
When it comes to Shelly and physical size, they have a new series of Mini devices, significantly smaller than the older Plus series, and good candidates for fitting in a box, as long as Wifi is not blocked too much by wires or other metal stuff.

**Reply by SylvainGa** [reply]
The mini is missing the extra button that I need to control through the other switch (three way).

**Reply by barbudor** [reply]
That would work with a Shelly mini
![image](https://github.com/arendst/Tasmota/assets/2996042/64e6e9b8-0168-4ef4-a1eb-d8919892fd74)

**Reply by SylvainGa** [reply]
> That would work with a Shelly mini
> ![image](https://github.com/arendst/Tasmota/assets/2996042/64e6e9b8-0168-4ef4-a1eb-d8919892fd74)
> 

True, by using two standard 3 way switches. I'll keep this in mind if the other solutions don't pan out. Thanks

**Comment by mecfiremike** [comment]
I have used these:
https://www.amazon.com/TASMOTA-Martin-Jerry-ESP8266-Assistant/dp/B0BJ9D1NM1/

There are other Martin Jerry 3-Way Pre-Flashed ones on there too if you are
looking for something with a dimmer.

My Preference if possible is to eliminate the 3-Way Circuit and just use a
Standard Martin Jerry Switch on switch location, and if you can get power
at the other location, add an additional standard smart switch that just
sends the MQTT Commands to the first switch.

**Reply by SylvainGa** [reply]
Yeah, I checked but it's $127 when in Canada 🙁

There is a 2-14 wire from the main box where the smart switch is to the remote switch location. I'll probably go this route.
